### PR TITLE
CAZ-1423 Validate chronological order of from/to dates in charge sett…

### PR DESCRIPTION
**Pull request by radekmorytko**
…lement API `payment-info` endpoint

### CODE REVIEW

File: src/main/java/uk/gov/caz/psr/dto/validation/constraint/FromAndToDatesInChronologicalOrderValidator.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ */
+public class FromAndToDatesInChronologicalOrderValidator implements
+    ConstraintValidator<FromAndToDatesInChronologicalOrder, PaymentInfoRequest> {
+
+  /**
+   * Checks if the passed {@code input} contains from and to dates in chronological order.
+   *
+   * @param input An input request which will be validated.
+   * @param context Validator context (unused).
+   * @return true if {@code input} contains at least on non-null parameter, false otherwise.
```
Comment (plenza) [inline src/main/java/uk/gov/caz/psr/dto/validation/constraint/FromAndToDatesInChronologicalOrderValidator.java]: Comment not updated

File: src/test/java/uk/gov/caz/psr/dto/validation/constraint/FromAndToDatesInChronologicalOrderValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import java.time.LocalDate;
+import javax.validation.ConstraintValidatorContext;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.caz.psr.dto.PaymentInfoRequest;
+
+@ExtendWith(MockitoExtension.class)
+class FromAndToDatesInChronologicalOrderValidatorTest {
```
Comment (plenza) [inline src/test/java/uk/gov/caz/psr/dto/validation/constraint/FromAndToDatesInChronologicalOrderValidatorTest.java]: Missing test for `from` and `to` equal - it should be OK if I've understood implementation correctly.

File: src/main/java/uk/gov/caz/psr/dto/validation/constraint/FromAndToDatesInChronologicalOrderValidator.java (hunk 1 above)
Comment (radekmorytko) [reply src/main/java/uk/gov/caz/psr/dto/validation/constraint/FromAndToDatesInChronologicalOrderValidator.java]: Fixed

File: src/test/java/uk/gov/caz/psr/dto/validation/constraint/FromAndToDatesInChronologicalOrderValidatorTest.java (hunk 2 above)
Comment (radekmorytko) [reply src/test/java/uk/gov/caz/psr/dto/validation/constraint/FromAndToDatesInChronologicalOrderValidatorTest.java]: You're right, I've added test cases.

**Pull request merged**